src/blocks/select/Select.tsx

```diff
@@ -0,0 +1,240 @@
+        border: none;
+        color: var(--components-inputs-text-default, ${colors['components-inputs-text-default']});
+
+        display: flex;
+
+        font-family: var(--font-family);
+        font-size: ${textVariants['bs-regular'].fontSize};
+        font-style: ${textVariants['bs-regular'].fontStyle};
+        font-weight: ${textVariants['bs-regular'].fontWeight};
+        line-height: ${textVariants['bs-regular'].lineHeight};
+
+        gap: var(--spacing-none, 0px);
+
+        &:focus {
+          outline: none;
+        }
+        &:hover {
+          outline: none;
+        }
+        &:disabled {
+          background-color: transparent;
+          cursor: not-allowed;
+          color: var(--components-inputs-text-disabled, ${colors['components-inputs-text-disabled']});
+        }
+
+        ::placeholder {
+          color: var(--components-inputs-text-placeholder, ${colors['components-inputs-text-placeholder']});
+        }
+      }
+    `;
+  }}
+`;
+
+const StyledPopover = styled(ComboboxPopover)<{ popoverWidth: number }>`
+  width: ${({ popoverWidth }) => popoverWidth}px;
+  margin: var(--spacing-xxs) var(--spacing-none) var(--spacing-none) var(--spacing-none);
+  padding: var(--spacing-xxs, 8px);
+  border-radius: var(--radius-xs, 12px);
+  border: var(--border-sm, 1px) solid var(--stroke-secondary, #eaebf2);
+  background: var(--surface-primary, #fff);
+  overflow: hidden auto;
+  max-height: 20rem;
+  position: absolute;
+`;
+
+const StyledCombobox = styled(Combobox)`
+  position: relative;
+  width: 100%;
+`;
+
+const StyledList = styled(ComboboxList)`
+  display: flex;
+  flex-direction: column;
+  gap: var(--spacing-xs, 12px);
+`;
+
+const StyledOption = styled(ComboboxOption)`
+  display: flex;
+  align-items: center;
+  padding: var(--spacing-xxxs, 4px);
+  gap: var(--spacing-xxs, 8px);
+  color: var(--components-dropdown-text-default, #17181b);
+  text-align: center;
+  font-family: var(--font-family);
+  font-size: ${textVariants['bs-regular'].fontSize};
+  font-style: ${textVariants['bs-regular'].fontStyle};
+  font-weight: ${textVariants['bs-regular'].fontWeight};
+  line-height: ${textVariants['bs-regular'].lineHeight};
+  &:hover {
+    border-radius: var(--radius-xxs, 8px);
+    background: var(--surface-secondary, #f5f6f8);
+  }
+  [role='img'] {
+    width: 24px;
+    height: 24px;
+  }
+`;
+
+const Select: React.FC<SelectProps> = ({
+  options,
+  onSelect,
+  css,
+  selectedValue,
+  placeholder = '',
+  error,
+  success,
+  disabled,
+}) => {
+  const [popoverWidth, setPopoverWidth] = useState(0);
+  const comboboxRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (comboboxRef.current) {
+        setPopoverWidth(comboboxRef.current.offsetWidth - 18);
+      }
+    };
+
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
```

Comment:
We need to use the same width of select input field for the popover. Let's achieve that us css if possible.